### PR TITLE
SRS-578: Implementing horizontal scrolling for minimized(mobile) version

### PR DESCRIPTION
**Pull request by nupurdixit13**
# Description

- The changes correspond to creating a horizontal scroller for mobile version. This is to make sure that when a user accesses the map on mobile, they have the ability to scroll horizontally left and right . Screenshots of how it looks with these changes have been attached for reference.
- Unit tests have been also included. 
Please note : Existing FE unit tests are failing for which Anton has already published a PR.

Screenshot 2024-10-24 at 3 16 37 PM
Screenshot 2024-10-24 at 3 16 44 PM
Screenshot 2024-10-24 at 3 16 51 PM

Please note: My latest commit only removes few unwanted variables in css. Since I pulled latest changes thus the count of files has been increased.

### CODE REVIEW

File: frontend/src/app/features/map/HorizontalScroller.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  //   const swipeCallback = (direction: SwipeDirection) => {
+  //     if (direction === 'left') {
+  //       onScrollRight();
+  //     } else if (direction === 'right') {
+  //       onScrollLeft();
+  //     }
+  //   };
+
+  //useSwipe(ref, swipeCallback)
+
```
Comment (jaspalsingh-aot) [inline frontend/src/app/features/map/HorizontalScroller.tsx]: Do we need this commented code?

File: frontend/src/app/features/map/HorizontalScroller.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
+import { ReactNode, useEffect, useRef, useState } from 'react';
+import { Button } from '@mui/material';
+import clsx from 'clsx';
+
+// import { SwipeDirection, useSwipe } from '../../../hooks/useSwipe';
+
```
Comment (jaspalsingh-aot) [inline frontend/src/app/features/map/HorizontalScroller.tsx]: do we need this commented code ?

File: frontend/src/app/features/map/HorizontalScroller.css [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+button.horizontal-scroller-button.horizontal-scroller-button--left {
+  left: 16px;
+}
+
+button.horizontal-scroller-button.horizontal-scroller-button--right {
+  right: 16px;
+}
+
```
Comment (jaspalsingh-aot) [inline frontend/src/app/features/map/HorizontalScroller.css]: can't we use design token, which we already have?

File: frontend/src/app/features/map/HorizontalScroller.css [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  align-items: center;
+  flex-wrap: nowrap;
+  overflow-y: visible;
+  overflow-x: hidden;
+}
+
+button.horizontal-scroller-button.MuiButton-root {
```
Comment (jaspalsingh-aot) [inline frontend/src/app/features/map/HorizontalScroller.css:17]: can you please make sure this won't override the bootstrap css ?

File: frontend/src/app/features/map/HorizontalScroller.css [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+  border-radius: 50%;
+  box-shadow: var(--box-shadow-small);
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  z-index: 2;
+}
+
+button.horizontal-scroller-button.horizontal-scroller-button--left {
+  left: 16px;
```
Comment (jaspalsingh-aot) [inline frontend/src/app/features/map/HorizontalScroller.css:39]: can you please make sure this won't override the bootstrap css ?

File: frontend/src/app/features/map/HorizontalScroller.css [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+button.horizontal-scroller-button.horizontal-scroller-button--right {
+  right: 16px;
```
Comment (jaspalsingh-aot) [inline frontend/src/app/features/map/HorizontalScroller.css:43]: same as above

**Comment by jaspalsingh-aot** [comment]
> # Description
> * The changes correspond to creating a horizontal scroller for mobile version. This is to make sure that when a user accesses the map on mobile, they have the ability to scroll horizontally left and right . Screenshots of how it looks with these changes have been attached for reference.
> * Unit tests have been also included.
>   Please note : Existing FE unit tests are failing for which Anton has already published a PR.
> 
> Screenshot 2024-10-24 at 3 16 37 PM Screenshot 2024-10-24 at 3 16 44 PM Screenshot 2024-10-24 at 3 16 51 PM


- The header appears to be a bit congested and doesn't align with the design specifications.
- It seems that the "Organic Info" section is currently missing from the implementation. 
- I noticed the absence of a language dropdown in actual design,  although included in the code .

I wanted to share a few considerations regarding the use of both libraries:

**Increased Bundle Size:** Integrating both libraries may lead to larger file sizes, which could potentially affect performance.

**Styling Conflicts:** There is a possibility of CSS conflicts or style overrides, which might result in unexpected behavior or appearance.

**Comment by jaspalsingh-aot** [comment]
do we need frontend test cases for polygon search, radius search and map search?

File: frontend/src/app/features/map/HorizontalScroller.tsx (hunk 1 above)
Comment (nupurdixit13) [reply frontend/src/app/features/map/HorizontalScroller.tsx]: Removed

File: frontend/src/app/features/map/HorizontalScroller.tsx (hunk 2 above)
Comment (nupurdixit13) [reply frontend/src/app/features/map/HorizontalScroller.tsx]: Removed

**Comment by nupurdixit13** [comment]
> > # Description
> > 
> > * The changes correspond to creating a horizontal scroller for mobile version. This is to make sure that when a user accesses the map on mobile, they have the ability to scroll horizontally left and right . Screenshots of how it looks with these changes have been attached for reference.
> > * Unit tests have been also included.
> >   Please note : Existing FE unit tests are failing for which Anton has already published a PR.
> > 
> > Screenshot 2024-10-24 at 3 16 37 PM Screenshot 2024-10-24 at 3 16 44 PM Screenshot 2024-10-24 at 3 16 51 PM
> 
> * The header appears to be a bit congested and doesn't align with the design specifications.
> * It seems that the "Organic Info" section is currently missing from the implementation.
> * I noticed the absence of a language dropdown in actual design,  although included in the code .
> 
> I wanted to share a few considerations regarding the use of both libraries:
> 
> **Increased Bundle Size:** Integrating both libraries may lead to larger file sizes, which could potentially affect performance.
> 
> **Styling Conflicts:** There is a possibility of CSS conflicts or style overrides, which might result in unexpected behavior or appearance.

As discussed, the abpve mentioned point do not correspond to this feature as the designs in general do comntain header from Organics Info which has no relevance with our feature

**Pull request closed without merging**